hack/jenkins/verify-jenkins-jobs.sh

```diff
@@ -72,5 +74,7 @@ e2e_builds=$(grep "^  kubernetes-.*)$" "${E2E}" | tr -d " )")
 jenkins_builds=$(curl -sg "${JENKINS}/api/json?tree=jobs[name]&pretty=true" \
   | grep -Po '(?<="name" : ")[^"]*')
 
-search_build "${e2e_builds}" "${jenkins_builds}" "Jenkins" \
-  || search_build "${jenkins_builds}" "${e2e_builds}" "e2e.sh"
+if ! search_build "${e2e_builds}" "${jenkins_builds}" "Jenkins" || \
```

Comment:
this now short-circuits such that the first check failing will prevent the second check from running, which isn't great either. ideally we'd run both checks and then fail if either one fails.